src/typescript/frontend/src/components/pages/metadata/MetadataPage.tsx

```diff
@@ -0,0 +1,276 @@
+const INPUT_CLASSNAME =
+  "text-white px-4 border-ec-blue border-solid border-[3px] rounded-xl bg-black outline-none";
+
+const FormEntry = ({
+  value,
+  onChange,
+  placeholder,
+  name,
+  onClick,
+  buttonText,
+  className,
+  disabled = false,
+}: {
+  value: string;
+  onChange: ChangeEventHandler<HTMLInputElement>;
+  onClick?: MouseEventHandler<HTMLButtonElement>;
+  placeholder: string;
+  name: string;
+  buttonText?: string;
+  className?: string;
+  disabled?: boolean;
+}) => (
+  <div className={`mx-auto relative`}>
+    <div className="max-w-[300px] text-white">{name}</div>
+    <input
+      tabIndex={1}
+      className={`${INPUT_CLASSNAME} w-[300px] ${className}`}
+      value={value}
+      placeholder={placeholder}
+      onChange={onChange}
+    ></input>
+    {buttonText && (
+      <button
+        className={`${BUTTON_CLASSNAME} absolute right-[-60px] ${disabled ? "!border-dark-gray" : ""}`}
+        onClick={onClick}
+        tabIndex={2}
+        disabled={disabled}
+      >
+        {buttonText}
+      </button>
+    )}
+  </div>
+);
+
+const MetadataPage = () => {
+  const [fields, setFields] = useState<Map<string, string>>(DEFAULT_FIELDS);
+  const [marketAddress, setMarketAddress] = useState<string>("");
+  const [addFieldName, setAddFieldName] = useState<string>("");
+  const [pasted, setPasted] = useState(false);
+
+  const { account, aptos, submit } = useAptos();
+
+  const isMarketAddressValid = useMemo(
+    () => AccountAddress.isValid({ input: marketAddress }).valid,
+    [marketAddress]
+  );
+
+  const { data: marketExists } = useQuery({
+    queryKey: ["metadata-market-exists", marketAddress],
+    queryFn: async () => {
+      if (!isMarketAddressValid) {
+        return null;
+      }
+      try {
+        const market = await MarketMetadataByMarketAddress.view({
+          aptos,
+          marketAddress,
+        });
+        return market.vec.length !== 0;
+      } catch (e) {
+        console.error("Could not get if market exists", e);
+        return false;
+      }
+    },
+  });
+
+  const marketAddressBorderColor = useMemo(() => {
+    switch (marketExists) {
+      case true:
+        return "border-green";
+      case false:
+        return "border-pink";
+      case null:
+      case undefined:
+        return "border-ec-blue";
+    }
+  }, [marketExists]);
+
+  const isSubmitEnabled = useMemo(
+    () => account !== null && marketExists === true,
+    [account, marketExists]
+  );
+
+  // Update fields on market address change
+  useEffect(() => {
+    // Don't fetch market properties if market address is not valid.
+    if (!isMarketAddressValid) {
+      return;
+    }
+    // Don't fetch market properties if user has pasted.
+    if (pasted) {
+      return;
+    }
+    MarketProperties.submit({
+      aptos,
+      market: marketAddress,
+    })
+      .then((r) => (r.vec.length === 0 ? null : (r.vec[0] ?? null)))
```

Comment:
```suggestion
      .then((r) => r.vec.at(0) ?? null)
```

Reply:
Fixed in f8a7353b87d46bad1fbfa010e10d2f0ab9ad9176.